File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt User for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentances in the text
    int letters = 0;
    int words = 1;
    int sentances = 0;

    // Compute the index
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentances++;
        }
    }

    float L = (float) letters / (float) words * 100;
    float S = (float) sentances / (float) words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print Grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: readability.py
from cs50 import get_string

text = get_string("Text: ")

letter = 0
space = 0
sentences = 0

for char in text:
    if char.isalpha():
        letter += 1
    elif char == ' ':
        space += 1
    elif char == '.' or char == '?' or char == '!':
        sentences += 1

words = space + 1

L = (letter / words) * 100
S = (sentences / words) * 100

grade = 0.0588 * L - 0.296 * S - 15.8

if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(grade)}")
